src/db/chainIndexers/getChainIndexer.ts

```diff
@@ -0,0 +1,27 @@
+import { PrismaTransaction } from "../../schema/prisma";
+import { getPrismaWithPostgresTx } from "../client";
+
+interface GetBlockForIndexingParams {
+  chainId: number;
+  pgtx?: PrismaTransaction;
+}
+
+export const getBlockForIndexing = async ({
+  chainId,
+  pgtx,
+}: GetBlockForIndexingParams) => {
+  const prisma = getPrismaWithPostgresTx(pgtx);
+
+  const lastIndexedBlock = await prisma.$queryRaw<
+    { lastIndexedBlock: number }[]
+  >`
+    SELECT
+      "lastIndexedBlock"
+    FROM
+      "chain_indexers"
+    WHERE
+      "chainId"=${chainId}
+    FOR UPDATE NOWAIT
```

Comment:
`FOR UPDATE NOWAIT` might give an issue/error, i.e., `Resource busy & cannot acquire ...` with multi-worker scenario

Reply:
This is expected, we should just catch the error

Reply:
got it